testsuite/tests/typing-core-bugs/missing_rec_hint.ml

```diff
@@ -33,15 +33,20 @@ and h x = if x < 0 then x else g (x-1)
 Line _, characters 31-32:
   let f x = if x < 0 then x else h (x-1)
                                  ^
-Error: Unbound value h.
-       Hint: You are probably missing the `rec' keyword on line 1.
+Error: Unbound value h
+Hint: If this is a recursive definition, you should add a 'rec' keyword on line 1
 |}];;
 
 let value1 = 3 in
-let value2 = value2 + 1 in  (* typo: should use value1 *)
+let value2 = value2 (* typo: should be value1 *) + 1 in
 ()
 
-[%expect{|
+[%%expect{|
+Line _, characters 13-19:
+  let value2 = value2 (* typo: should be value1 *) + 1 in
+               ^^^^^^
+Error: Unbound value value2
+Hint: Did you mean value1?
```

Comment:
At this point in the patch series, there is no heuristic to disable the missing-rec handling for non-functions (added in the next commit  92ff83e4022c05753bcbd0630905db40708eae65 ). How come there is no recursive-definition here, does the spellchecking hint override it naturally?

Reply:
No, it appears I just messed up the tests while rebasing. This should be fixed now.